### PR TITLE
Prevent Viewer Panel Resize Crashing Whiteboard

### PR DESCRIPTION
**Pull request by KDSBrowne**
### What does this PR do?
Fixes the whiteboard crashing when a viewer resizes a panel.

### CODE REVIEW

File: bigbluebutton-html5/imports/ui/components/whiteboard/component.jsx [hunk 1]
```diff
@@ -206,271 +206,299 @@ export default function Whiteboard(props) {
...
+    <>
+      <Cursors
+        tldrawAPI={tldrawAPI}
+        currentUser={currentUser}
+        hasMultiUserAccess={props?.hasMultiUserAccess}
+        whiteboardId={whiteboardId}
+        isViewersCursorLocked={isViewersCursorLocked}
+        isMultiUserActive={isMultiUserActive}
+      >
+        {!hasWBAccess && !isPresenter ? readOnlyWB : editableWB}
```
Comment (antobinary) [inline bigbluebutton-html5/imports/ui/components/whiteboard/component.jsx:501]:
```suggestion
        {!hasWBAccess && !isPresenter ? readOnlyWB : editableWB}
        {hasWBAccess || isPresenter ? editableWB : readOnlyWB}
```

Just an idea

File: bigbluebutton-html5/imports/ui/components/whiteboard/component.jsx [hunk 2]
```diff
@@ -206,271 +206,299 @@ export default function Whiteboard(props) {
...
+    if (reason && isPresenter && (reason.includes("zoomed") || reason.includes("panned"))) {
+      if (cameraFitSlide.zoom === 0) {
+        //can happen when the slide finish uploading
+        const cameraFitSlide = calculateCameraFitSlide();
+        tldrawAPI?.setCamera(cameraFitSlide.point, cameraFitSlide.zoom);
+        setIsZoomed(false);
+        setCameraFitSlide(cameraFitSlide);
+        return;
+      }
+      const camera = tldrawAPI.getPageState().camera;
```
Comment (antobinary) [inline bigbluebutton-html5/imports/ui/components/whiteboard/component.jsx:260]: is there a case where `tldrawAPI.getPageState()` is null or undefined?

File: bigbluebutton-html5/imports/ui/components/whiteboard/component.jsx [hunk 3]
```diff
@@ -206,271 +206,299 @@ export default function Whiteboard(props) {
...
+      } else {
+        tldrawAPI?.setCamera([slidePosition.xCamera, slidePosition.yCamera], slidePosition.zoom);
+        setIsZoomed(true);
+      }
+    }
+  };
+
+  const editableWB = (
+    <Tldraw
+      key={`wb-${!hasWBAccess && !isPresenter}-${document?.documentElement?.dir}-${document.getElementById('Navbar')?.style?.width}`}
```
[the remark this answers is not in the document]
Comment (KDSBrowne) [reply bigbluebutton-html5/imports/ui/components/whiteboard/component.jsx]: This key is using the document direction and nav width to force an update when they change. They are needed for the cursor updates.

File: bigbluebutton-html5/imports/ui/components/whiteboard/component.jsx (hunk 2 above)
Comment (KDSBrowne) [reply bigbluebutton-html5/imports/ui/components/whiteboard/component.jsx]: updated the operator in the event it's `null` or `undefined`

**Review by ramonlsouza — APPROVED** [review]

**Pull request merged**